solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java

```diff
@@ -384,12 +430,12 @@ public boolean equals(Object o) {
 			if (this == o) return true;
 			if (o == null || getClass() != o.getClass()) return false;
 			FlowReceiverReference that = (FlowReceiverReference) o;
-			return id.equals(that.id) && flowReceiver.equals(that.flowReceiver);
+			return Objects.equals(id, that.id) && Objects.equals(flowReceiver, that.flowReceiver) && Objects.equals(transactedSession, that.transactedSession);
```

Comment:
equals can be simplified, is is sufficient

Reply:
Fixed to just check the `id` for both these methods.